crates/mem-pool/src/sync/fan_out.rs

```diff
@@ -0,0 +1,76 @@
+use gw_types::{
+    offchain::DepositInfo,
+    packed::{
+        BlockInfo, L2Transaction, NextMemBlock, RefreshMemBlockMessageUnion, WithdrawalRequest,
+    },
+    prelude::{Builder, Entity, Pack, PackVec},
+};
+use smol::channel::{Receiver, Sender};
+
+use super::mq::{gw_kafka, Produce};
+
+const CHANNEL_BUFFER_SIZE: usize = 500;
+pub(crate) struct FanOutMemBlockActor {
+    receiver: Receiver<RefreshMemBlockMessageUnion>,
+    producer: gw_kafka::Producer,
+}
+
+impl FanOutMemBlockActor {
+    pub(crate) fn new(
+        receiver: Receiver<RefreshMemBlockMessageUnion>,
+        producer: gw_kafka::Producer,
+    ) -> Self {
+        Self { receiver, producer }
+    }
+
+    fn handle(&mut self, msg: RefreshMemBlockMessageUnion) {
+        if let Err(err) = self.producer.send(msg) {
+            log::error!("[Fan out mem block] message failed: {:?}", err);
+        }
+    }
+}
+
+async fn fan_out_handle(mut actor: FanOutMemBlockActor) {
+    log::info!("Fanout handle is started.");
+    while let Ok(msg) = actor.receiver.recv().await {
```

Comment:
1. we should output the error from recv. Do not ignore any errors.
2. we should exit the actor / task if channel is ended, you can detect that from the error.